tests/io/vasp/test_outputs.py

```diff
@@ -1736,6 +1819,32 @@ def test_standard(self):
         finally:
             sys.stdout = saved_stdout
 
+    def test_vaspwave(self):
+        wavecar = self.wavecar_fccSi
+        vaspwave = self.vaspwave_fccSi
+        assert vaspwave.filename == f"{VASP_OUT_DIR}/vaspwave.fccSi.h5.gz"
+        assert vaspwave.efermi == approx(wavecar.efermi, abs=1e-4)
+        assert vaspwave.encut == wavecar.encut
+        assert vaspwave.nb == wavecar.nb
+        assert vaspwave.nk == wavecar.nk
+        assert_allclose(vaspwave.a, wavecar.a)
+        assert_allclose(vaspwave.b, wavecar.b)
+        assert vaspwave.vol == approx(wavecar.vol)
+        assert len(vaspwave.kpoints) == wavecar.nk
+        assert len(vaspwave.coeffs) == wavecar.nk
+        assert len(vaspwave.coeffs[0]) == wavecar.nb
+        assert len(vaspwave.band_energy) == wavecar.nk
+        assert vaspwave.band_energy[0].shape == (wavecar.nb, 3)
+        for k in range(vaspwave.nk):
+            for b in range(vaspwave.nb):
+                assert len(vaspwave.coeffs[k][b]) == len(wavecar.Gpoints[k])
+
+        with pytest.raises(ValueError, match="invalid filetype='json'"):
+            Wavecar(f"{VASP_OUT_DIR}/vaspwave.fccSi.h5.gz", filetype="json", vasp_type="std")
+
+        with pytest.raises(ValueError, match="Invalid rtag=.+, must be one of"):
```

Comment:
Default filetype is legacy. It will result in Value error for in valid rtag while reading hdf5 file as it tries to read the hdf5 file as legacy WAVECAR file. So, hdf5 filetype should be provided while reading hdf5 file.